tests/context_tests.rs

```diff
@@ -0,0 +1,1147 @@
+    0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+    0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+    0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+    0, 0, 0, 0, 0,
+];
+
+use std::convert::TryInto;
+use tss_esapi::constants::*;
+use tss_esapi::tss2_esys::*;
+use tss_esapi::utils::{
+    self, AsymSchemeUnion, ObjectAttributes, PublicIdUnion, PublicParmsUnion, Signature,
+    Tpm2BPublicBuilder, TpmaSession, TpmsRsaParmsBuilder, TpmtSymDefBuilder,
+};
+use tss_esapi::*;
+
+#[test]
+fn simple_test() {
+    env_logger::init();
+
+    let mut context = Context::new(Tcti::Mssim).unwrap();
+    let key_auth: Vec<u8> = context.get_random(16).unwrap();
+
+    let prim_key_handle = context
+        .create_primary_key(
+            ESYS_TR_RH_OWNER,
+            &utils::get_rsa_public(true, true, false, 2048),
+            &key_auth,
+            &[],
+            &[],
+            &[],
+        )
+        .unwrap();
+
+    dbg!(prim_key_handle);
+
+    let new_session = context
+        .start_auth_session(
+            NO_SESSIONS,
+            ESYS_TR_NONE,
+            prim_key_handle,
+            &[],
+            TPM2_SE_HMAC,
+            utils::TpmtSymDefBuilder::aes_256_cfb(),
+            TPM2_ALG_SHA256,
+        )
+        .unwrap();
+    let session_attr = TpmaSession::new()
+        .with_flag(TPMA_SESSION_DECRYPT)
+        .with_flag(TPMA_SESSION_ENCRYPT);
+    context.set_session_attr(new_session, session_attr).unwrap();
+    context.set_sessions((new_session, ESYS_TR_NONE, ESYS_TR_NONE));
+
+    let (key_priv, key_pub) = context
+        .create_key(
+            prim_key_handle,
+            &utils::get_rsa_public(false, false, true, 1024),
+            &key_auth,
+            &[],
+            &[],
+            &[],
+        )
+        .unwrap();
+    let key_handle = context.load(prim_key_handle, key_priv, key_pub).unwrap();
+    dbg!(key_handle);
+
+    let key_context = context.context_save(key_handle).unwrap();
+    let key_handle = context.context_load(key_context).unwrap();
+    context.set_handle_auth(key_handle, &key_auth).unwrap();
+    dbg!(key_handle);
+    let scheme = TPMT_SIG_SCHEME {
+        scheme: TPM2_ALG_NULL,
+        details: Default::default(),
+    };
+    let validation = TPMT_TK_HASHCHECK {
+        tag: TPM2_ST_HASHCHECK,
+        hierarchy: TPM2_RH_NULL,
+        digest: Default::default(),
+    };
+    let signature = context
+        .sign(key_handle, &HASH[..32], scheme, &validation)
+        .unwrap();
+    print!("Signature: ");
+    for x in &signature.signature {
+        print!("{}, ", x);
+    }
+    println!();
+    dbg!(
+        context
+            .verify_signature(key_handle, &HASH[..32], &signature.try_into().unwrap())
+            .unwrap()
+            .tag
+    );
+}
+
+mod test_start_sess {
```

Comment:
Did you create a new module to launch those tests separately from the one above?

Reply:
Could all tests be under:
```
#[cfg(test)]
mod tests {
    #[test]
    fn test() {
        ...
    }
}
```
(same way as we do in Parsec)

Reply:
I did make a module per function that's tests, just to group them a bit, don't know if having all of them in one module makes any difference.
Also, the `cfg` isn't needed since this is a test file so it's by default only checked for tests

Reply:
Fair enough!